decidim-admin/app/commands/decidim/admin/suspend_user.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    class SuspendUser < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form - SuspendUserForm
+      def initialize(form)
+        @form = form
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the resource.
+      # - :invalid if the resource is not reported
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) unless form.valid?
+
+        transaction do
+          suspend!
+          register_justification!
+          notify_user!
+        end
+
+        broadcast(:ok, form.user)
+      end
+
+      private
+
+      attr_reader :form
+
+      def register_justification!
+        @current_suspension = UserSuspension.create!(
+          justification: form.justification,
+          user: form.user,
+          suspending_user: form.current_user
+        )
+      end
+
+      def notify_user!
+        Decidim::Admin::UserSuspensionJob.perform_later(@current_suspension.suspending_user,
+                                                        @current_suspension.user,
+                                                        @current_suspension.justification)
+      end
+
+      def suspend!
+        Decidim.traceability.perform_action!(
+          "suspend",
+          form.user,
+          form.current_user,
+          extra: {
+            reportable_type: form.user.class.name,
+            current_justification: form.justification
+          }
+        ) do
+          form.user.suspended = true
+          form.user.suspended_at = Time.current
+          form.user.suspension = @current_suspension
+          form.user.extended_data["user_name"] = form.user.name
+          form.user.name = "Blocked user"
```

Comment:
Maybe this should be translated?

Reply:
@oriolgual I am not sure about this, in general the user name is not translated.

Reply:
I agree with @oriolgual. But I think we should not rename the user. Instead I'll render a different user name if the user is blocked. This change will required adding a filter for blocked users in the participants admin panel.

@decidim/product what's your point of view here?